analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -484,7 +484,7 @@ private data class ModuleInfo(
     val main: Boolean = false,
 
     @JsonProperty("GoMod")
-    val goMod: String
+    val goMod: String?
```

Comment:
Just checking: In @oheger-bosch's [proposal](https://github.com/oss-review-toolkit/ort/pull/6638/files#diff-d0e6f488a95a59913828fe32b76dae1b1ad197ddac08af8485ffa87578a0ae79R489) this also got a default value of `null`. Is this intentionally *not* done here so somehow express that `goMod` being `null` is rather an exceptional case than the default?

Reply:
This class is only construction in context of deserialization. There is no constructor calls in the code. So, a default null value is not needed.

Reply:
Now I'm confused: Isn't the point that the `goMod` property might be missing from the JSON to deserialize, hence we'd need a default value here?

Reply:
Ah, seems like so. I've never made that really clear to me, it seems.

Given that in all other places where data class properties are nullable a default value of `null` *is* assigned, how about also aligning this as part of this PR? (In either way.)